source/kraken/realtime.cpp

```diff
@@ -250,17 +250,23 @@ get_status(const transit_realtime::TripUpdate_StopTimeEvent& event,
     }
 }
 
-static bool is_added_service(const transit_realtime::TripUpdate& trip_update) {
+static bool is_added_trip(const transit_realtime::TripUpdate& trip_update) {
     namespace trt = transit_realtime;
     auto log = log4cplus::Logger::getInstance("realtime");
-    using nt::disruption::StopTimeUpdate;
 
-    // adding a trip is adding service
     if (trip_update.trip().schedule_relationship() == trt::TripDescriptor_ScheduleRelationship_ADDED) {
         LOG4CPLUS_TRACE(log, "Disruption has ADDITIONAL_SERVICE effect");
         return true;
     }
-    else if (trip_update.trip().schedule_relationship() == trt::TripDescriptor_ScheduleRelationship_SCHEDULED
+    return false;
+}
+
+static bool is_added_stop_time(const transit_realtime::TripUpdate& trip_update) {
```

Comment:
We must keep the fact that adding trip is adding stop_time.
So the the final `return false` should probably be `return is_added_trip`.

Reply:
yes indeed. But this function only test if we have `adding stop_time + Scheduled`. 
It corresponds to the previous case (A simple adding stop time without new trip)
So, the function return `true` only if we detect a `ADDED/ADDED_FOR_DETOUR stop time`.

Reply:
Then we won't use it directly, I guess.
To me, especially on something as touchy as what we're trying to do: the safest is the best.

Being atomic is nice, but the main goal is to be safe (both is better).

If we ever get a trip add with only updates in it (if kirin ever sends it) we must not handle the feed. And there is no guarantee on Kirin (or even COTS) from doing that anytime.

Reply:
For this point, I don't understand your thought. The code is written in such a way that, if you have _is_realtime_add_enabled=false_, we'll not handle this kind of COTS files (add atop time)

with 
```
 if (! is_realtime_add_enabled && is_added_stop_time(trip_update)) {
￼         LOG4CPLUS_DEBUG(log, "Add stop time is disabled: ignoring it");
￼         return;
}
```

Reply:
After the call, now we want to combine flags logics to create priority

Reply:
Good for me : combination handled earlier, it's fine either way in that part of the code, you can separate logic :+1: